lgn-messages/src/types/v1/preprocessing/ext_tasks.rs

```diff
@@ -292,15 +298,23 @@ impl FinalExtraction {
 
 /// Inputs for a single table proof.
 ///
-/// This can be either a simple valued or a mapping table.
+/// # Identifiers
+///
+/// A [SingleTableExtraction] is either a final or a sub extraction. Final extractions can be
+/// either a simple table, a mapping table, or a mapping table with length. A sub extraction is
+/// either a simple or a no-length mapping table, which is later used by a merge extraction.
+///
+/// When this extraction is final, `table_id` must match the table's unique identifier. Sub
+/// extractions on the other hand should use the `table_hash`.
+///
+/// The `table_hash` is used to match identify the value extraction.
 #[derive(Clone, Dbg, PartialEq, Deserialize, Serialize)]
 pub struct SingleTableExtraction {
     pub table_id: TableId,
+    pub table_hash: TableHash,
+    pub value_proof_version: MptNodeVersion,
```

Comment:
Value extractions were previously identified with `(table_id, last_changed_block_nr, state_root)`. With merge tables, this is no longer unique, so the ID was changed to `(table_hash, last_changed_block_nr, state_root)`.